sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java

```diff
@@ -175,6 +176,11 @@ public void invokeOnTimer(
                 timerId, delegate.getClass().getName()));
       }
     }
+
+    @Override
+    public Serializable getFn() {
```

Comment:
I think you can make this `DoFn` and just crash if it is an `OldDoFn`. The direct runner no longer directly uses `OldDoFn` anyhow.